### PR TITLE
v1.0 release

### PR DESCRIPTION
**Pull request by quincy-kh-chen**
Migrate TRI internal DGP codebase including major components:
- Protos
- DatasetClass (Synchronized, Frame)
- CLI with visualizer

---
This change is [Reviewable](https://reviewable.io/reviews/tri-ml/dgp/18)

### CODE REVIEW

**Review by quincy-kh-chen — COMMENTED** [review]
+@ytsuzuki for review FYI @kuanleetri @wadimkehl 

> *[Reviewable](https://reviewable.io/reviews/tri-ml/dgp/18#-:-MkK6WJT2xSqKKw6lRyW:b714h73)* status: 0 of 82 files reviewed, all discussions resolved (waiting on @ytsuzuki)

**Review by quincy-kh-chen — COMMENTED** [review]
-@ytsuzuki +@yuta-tsuzuki-woven for review

> *[Reviewable](https://reviewable.io/reviews/tri-ml/dgp/18#-:-MkKOHbt8cw4IXT4GFVx:b-8ozvp9)* status: 0 of 82 files reviewed, all discussions resolved (waiting on @yuta-tsuzuki-woven)

**Review by yuta-tsuzuki-woven — APPROVED** [review]
@quincy-kh-chen 
Thank you for this PR!
I checked proto files and `utils/` modules and they are LGTM.
If possible, it would be nice to see the differences between this repository and the ouroboros on stargate, but if there are no differences, please ignore this comment.

> Reviewed 69 of 74 files at r1, 1 of 2 files at r2, 1 of 6 files at r3, 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tri-ml/dgp/18#-:-MkNjJtGBENNSV4_EsD-:bg7gloh)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @quincy-kh-chen)

**Review by quincy-kh-chen — COMMENTED** [review]
+documentation 

> *[Reviewable](https://reviewable.io/reviews/tri-ml/dgp/18#-:-MkPZ6te35Sjqih6XLvE:b-5jjjtg)* status: 76 of 85 files reviewed, all discussions resolved (waiting on @yuta-tsuzuki-woven)

File: dgp/proto/dataset.proto [hunk 1]
```diff
@@ -37,19 +39,23 @@ enum DatasetSplit {
...
 message DatasetMetadata {
   // Dataset name
   string name = 1;
 
   // Dataset semantic version
+  // TODO: Establish semantic versioning.
   string version = 2;
 
   // Date of dataset creation: yyyy-mm-dd
+  // TODO: Change this to datetime
```
Comment (wadimkehl) [inline dgp/proto/dataset.proto:58]: Can we robustify this to UTC? We will have data from all around the globe now

File: dgp/proto/dataset.proto [hunk 2]
```diff
@@ -37,19 +39,23 @@ enum DatasetSplit {
   TRAIN = 0;
   VAL = 1;
   TEST = 2;
+
+  // TODO (allan.raventos): deprecate this once we have a stable version of the DGP tagged
```
Comment (wadimkehl) [inline dgp/proto/dataset.proto:43]: Is this TODO still relevant?

File: dgp/utils/pose.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
+"""General-purpose class for rigid-body transformations.
+"""
+import numpy as np
+from pyquaternion import Quaternion
```
Comment (wadimkehl) [inline dgp/utils/pose.py:4]: We might want to switch to scipy quaternions at some point:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html

File: dgp/utils/pose.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
+"""General-purpose class for rigid-body transformations.
```
Comment (wadimkehl) [inline dgp/utils/pose.py:1]: Copyright?

File: dgp/utils/protobuf.py [hunk 5]
```diff
@@ -1,14 +1,16 @@
-# Copyright 2019 Toyota Research Institute.  All rights reserved.
+import hashlib
```
Comment (wadimkehl) [inline dgp/utils/protobuf.py:1]: Copyright?

File: dgp/utils/structures/bounding_box_2d.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
+import hashlib
```
Comment (wadimkehl) [inline dgp/utils/structures/bounding_box_2d.py:1]: Copyright?

File: dgp/utils/structures/bounding_box_3d.py [hunk 7]
```diff
@@ -0,0 +1,305 @@
+import hashlib
```
Comment (wadimkehl) [inline dgp/utils/structures/bounding_box_3d.py:1]: Copyright?

File: dgp/utils/structures/instance_mask.py [hunk 8]
```diff
@@ -0,0 +1,121 @@
+import hashlib
```
Comment (wadimkehl) [inline dgp/utils/structures/instance_mask.py:1]: Copyright?

File: dgp/utils/structures/key_point_2d.py [hunk 9]
```diff
@@ -0,0 +1,93 @@
+import hashlib
```
Comment (wadimkehl) [inline dgp/utils/structures/key_point_2d.py:1]: Copyright?

**Review by wadimkehl — COMMENTED** [review]
Great work, guys! :) Just have a couple of small items.

**Review by quincy-kh-chen — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tri-ml/dgp/18)* status: 76 of 90 files reviewed, 9 unresolved discussions (waiting on @wadimkehl and @yuta-tsuzuki-woven)

___
*[dgp/proto/dataset.proto, line 43 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVxRMLfPMKGVlP9-r9-43:-MkiHVUq9rUSE7wx4vPP:b-d2s0ud) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/proto/dataset.proto#L43)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

Is this TODO still relevant?
</blockquote></details>

Decided to keep `TRAIN_OVERFIT`. Removed TODO.
___
*[dgp/proto/dataset.proto, line 58 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVxRMLfPMKGVlP9-r9-58:-MkiHcjh2_RMkSIIK5Py:bfsor5a) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/proto/dataset.proto#L58)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

Can we robustify this to UTC? We will have data from all around the globe now
</blockquote></details>

Great suggestion! Let's keep it as a string now to ensure backward compatibility so that we can load existing DGP datasets.
We actually consider re-designing how we fetch and version the dataset. The dataset version/creation_date might be removed from the Metadata.
___
*[dgp/utils/pose.py, line 1 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVy8D7gCOjyNDhj-r9-1:-MkiJF4pBXkeDTb6ZRA_:b-896fix) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/utils/pose.py#L1)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

Copyright?
</blockquote></details>

Done.
___
*[dgp/utils/pose.py, line 4 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVy8D7gCOjyNDhj-r9-4:-MkiJGmQ2VZtaH8suOiu:b-896fix) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/utils/pose.py#L4)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

We might want to switch to scipy quaternions at some point:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html
</blockquote></details>

Done.
___
*[dgp/utils/protobuf.py, line 1 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVy8D7gCOjyNDhk-r9-1:-MkiJIckESBoMeHMehYm:b-896fix) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/utils/protobuf.py#L1)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

Copyright?
</blockquote></details>

Done.
___
*[dgp/utils/structures/bounding_box_2d.py, line 1 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVy8D7gCOjyNDhm-r9-1:-MkiJJP-AsgLq1m3DJQC:b-896fix) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/utils/structures/bounding_box_2d.py#L1)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

Copyright?
</blockquote></details>

Done.
___
*[dgp/utils/structures/bounding_box_3d.py, line 1 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVy8D7gCOjyNDhn-r9-1:-MkiJK1jAuwZntNKk0Ve:b-896fix) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/utils/structures/bounding_box_3d.py#L1)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

Copyright?
</blockquote></details>

Done.
___
*[dgp/utils/structures/instance_mask.py, line 1 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVy8D7gCOjyNDho-r9-1:-MkiJKl65cw5me3XkCiH:b-896fix) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/utils/structures/instance_mask.py#L1)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

Copyright?
</blockquote></details>

Done.
___
*[dgp/utils/structures/key_point_2d.py, line 1 at r9](https://reviewable.io/reviews/tri-ml/dgp/18#-Mk8pxVy8D7gCOjyNDhp-r9-1:-MkiJLHt-pUPphUvwPPR:b-896fix) ([raw file](https://github.com/tri-ml/dgp/blob/2a0d90b43d0ff0f98430b3ca8f600787d21c6ea9/dgp/utils/structures/key_point_2d.py#L1)):*
<details><summary><i>Previously, wadimkehl (Wadim Kehl) wrote…</i></summary><blockquote>

Copyright?
</blockquote></details>

Done.

**Comment by quincy-kh-chen** [comment]
@yuta-tsuzuki-woven @wadimkehl I open a new PR #20 with a squashed commit to replacing this PR. Also, add more documentations and resolve some comments from this PR. No change to protos and utils.

**Pull request closed without merging**